cpp/mg_utility/mg_test_utils.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+
+#include <chrono>
+
+namespace mg_test_utility {
+/// This class is threadsafe
+class Timer {
+ public:
+  explicit Timer() : start_time_(std::chrono::steady_clock::now()) {}
+
+  template <typename TDuration = std::chrono::duration<double>>
+  TDuration Elapsed() const {
+    return std::chrono::duration_cast<TDuration>(std::chrono::steady_clock::now() - start_time_);
+  }
+
+ private:
+  std::chrono::steady_clock::time_point start_time_;
```

Comment:
I assume there is not use case to ever change the start_time, since its being set in constructor
```suggestion
  const std::chrono::steady_clock::time_point start_time_;
```

Reply:
I think it's generally accepted that having const member variables are more trouble than they're worth.
First of all, it disables move, because const values cannot be moved (this is why you should never return a const value from a function).
Second, you cannot use copy assignment as it copies field by field, so you cannot copy the start_time from one timer into another.
You can create API around that, but it's better to have a simple nonconst private member and control the access using const getters/methods.